tests/test_measurements.py

```diff
@@ -435,7 +435,7 @@ def test_measurement_basis_list(backend):
     c.add(gates.H(2))
     c.add(gates.X(3))
     c.add(gates.M(0, 1, 2, 3, basis=[gates.X, gates.Z, gates.X, gates.Z]))
-    result = c(nshots=100)
+    result = backend.execute_circuit(c, nshots=100)
```

Comment:
I like a lot the explicit backend usage, but why do you need it for this PR?

Reply:
Because otherwise only the `GlobalBackend` is tested here, if I am not mistaken. Anyway, this was failing with the `QulacsBackend`, but was not supposed to be testing it, even though it did due to the `GlobalBackend` usage.

Reply:
> I actually searched, and only a few tests are actually using `set_backend()`, most of them just to test it.

This is fine, but there are also tests that are using `circuit()` without `set_backend()`. These are also invoking the `GlobalBackend` to execute the circuit and are even worse, because the `GlobalBackend` will be whetever was set beforehand (by the last `set_backend()` of a previous test). These may also be a bit harder to search for, because they can appear in various ways: `circuit()`, `circuit.execute()`, `c()`, `whatever_circuit_name()`, `c(nshots=100)` (like the one fixed here), etc.. An easy way would be to remove `Circuit.__call__` and `Circuit.execute` methods and see which tests fail.

Reply:
> By the way, I think that this discussion is not very relevant to this PR and maybe should be converted to an issue.

I agree about the issue. If you wish, feel free to open yourself, you're definitely the most expert.

> An easy way would be to remove `Circuit.__call__` and `Circuit.execute` methods and see which tests fail.

Another option is to keep `Circuit.__call__`, but require a `backend` argument to be passed.

Reply:
> I agree about the issue. If you wish, feel free to open yourself, you're definitely the most expert.

#1374

Reply:
Do we have a list of tests relying on a random state of the `GlobalBackend`?

Reply:
> Do we have a list of tests relying on a random state of the `GlobalBackend`?

These are the tests that execute circuits using the `circuit.execute` interface, which invokes the `GlobalBackend()`:

* `test_backends_global::test_circuit_execution` (sets to numpy)
* `test_measurements::test_measurement_compiled_circuit` (does not set)
* `test_models_circuit_execution::test_compiled_execute` (does not set)
* Various tests in `test_models_qcnn` are using the `GlobalBackend` but setting it beforehand to qibojit or numpy without reseting it after.
* `test_models_variational::test_vqe`  (does not set)
* `test_models_variational::test_custom_loss` (does not set)